rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf

```diff
@@ -1154,8 +1163,9 @@ SecRule REQUEST_HEADERS_NAMES "@rx ^.*$" \
     severity:'CRITICAL',\
     setvar:'tx.header_name_%{tx.0}=/%{tx.0}/',\
     chain"
-    SecRule TX:/^header_name_/ "@within %{tx.restricted_headers}" \
+    SecRule TX:/^header_name_/ "@within %{tx.restricted_headers_basic}" \
```

Comment:
How does this work?? This is horrible. There's no documentation on this "feature". It would be much simpler to use the rule ID as prefix for the variable, if the intention is to prevent clashes.

Reply:
This is a re-use of the existing code/rule. The existing rule uses a regular expression in the target, which is a documented engine feature [(reference manual link)](https://github.com/SpiderLabs/ModSecurity/wiki/Reference-Manual-(v2.x)#secruleupdatetargetbyid). The CRS documentation and the ModSecurity Handbook do refer to / document it, too.

I think if we want to consider changing that mechanism, that's definitely outside the scope of this issue and PR. But it's worthy of consideration, if there is a better solution, IMO.

Reply:
Okay. `headers_basic` and `headers_extended` was proposed in the original issue, which is why I used it. I'll put forward your suggestion in the original thread and see what people think.

Reply:
The former (`SecRule TX:/^header_name_/`).

Reply:
Thinking about this again, I don't like the idea of doing the
`setvar:'tx.header_name_%{tx.0}=/%{tx.0}/',\`
step twice.

I tried to take the simplest and the most uncontroversial approach here: simply copy and paste the existing rule, no new mechanisms, just the same logic repeated for the new "extended" headers.

But, thinking about it, I can't see any good reason to repeat the `/slashing/` process and setting all those variables *twice*.

Surely the original mechanism at PL 1 can be left intact, and then the *new* rule at PL 2 can just re-inspect all of those `/slashed/` variables that have already been set… No need to set them twice, I think.

Reply:
Actually, no, we can't do that. :sweat_smile: 

We cannot assume that the earlier PL 1 rule has already taken care of transforming the request header names into lowercase and wrapping them in `/slashes/`. It's possible that a user would tune away FPs and `SecRuleRemoveById` the PL 1 rule, and then the PL 2 "extended" rule would be broken.

Reply:
Okay: two sets of transformed request header names prefixed by the rule IDs are _in_. Let me know what you think, when you have a moment :)